app/code/core/Mage/Log/etc/config.xml

```diff
@@ -29,9 +29,24 @@
             </entities>
         </ignoredModules>
         <ignore_user_agents>
-            <google1>Googlebot/1.0 (googlebot@googlebot.com http://googlebot.com/)</google1>
-            <google2>Mozilla/5.0 (compatible; Googlebot/2.1; +http://www.google.com/bot.html)</google2>
-            <google3>Googlebot/2.1 (+http://www.googlebot.com/bot.html)</google3>
+            <google>Googlebot</google>
+            <mauibot>MauiBot</mauibot>
+            <blexbot>BLEXBot</blexbot>
+            <hrefsbot>AhrefsBot</hrefsbot>
+            <bingbot>bingbot</bingbot>
+            <dotbot>DotBot</dotbot>
+            <istellabot>istellabot</istellabot>
+            <mod_pagespeed>mod_pagespeed</mod_pagespeed>
+            <mj12bot>MJ12bot</mj12bot>
+            <yandexbot>YandexMobileBot</yandexbot>
+            <yandeximage>YandexImages</yandeximage>
+            <opensiteexplorer>spbot</opensiteexplorer>
+            <semrushbot>SemrushBot</semrushbot>
+            <alphabot>AlphaBot</alphabot>
+            <facebookexternalhit>facebookexternalhit</facebookexternalhit>
+            <seotester>SEOTesterBot</seotester>
+            <ZoominfoBot>ZoominfoBot</ZoominfoBot>
+            <amazonbot>Amazonbot</amazonbot>
```

Comment:
Where do you get the list? Is it better to order it alphabetically?

Reply:
> Where do you get the list?

Also searched for something like this. Maybe adopt https://github.com/fabiomb/is_bot?

Reply:
Also, is it a good idea to ignore **every** crawlers/bots, or **only** the popular ones? I do not know how many bots are out there, could be millions. I tend to think it's better to ignore popular bots, and log the rest. The log data could show anomaly patterns that the site owner can take certain actions. Does it make sense?

Reply:
> Where do you get the list? Is it better to order it alphabetically?

It's a list that I have built over time, the list of bots can be endless and never definitive. It could also be added to the configuration.

Reply:
I think Sven's finding on `is_bot` is a good suggestion. Community maintained list is better.

Reply:
See also https://github.com/JayBizzle/Crawler-Detect

Reply:
the is_bot does it in the way of this PR, but has a big list and before in the comments we were saying "if the list gets big..."

is_bot is GPL2 which is not compatible with OSL/AFL

Crawler-Detect works with regex and the list is very big

maybe we could have a long string instead of an array, to make things simple?

Reply:
we can obtain a string for a regexp like "/google|bing|amazon/" imploding ignore agents array like:
```
$ignoreAgents = $ignoreAgents->asArray();
$ignoreAgents = implode("|", $ignoreAgents);
$isBot = (bool) preg_match("/{$ignoreAgents}/i", $userAgent);
```
we are sure regexp is slower then a foreach over stripos? maybe because exit on the first match?

Anyway, just searching for the main bots string would already make a big difference, and we have the same problem with sessions, which is more serious because sessions initiated by unidentified bots grow rapidly. In fact, I wanted to open a separate PR for it.

Reply:
Made some tests, but using `preg_match` changes behavoir. :/

Reply:
What do you mean by changes behavior? Are you referring to performance? Here's an example regex that I could build
https://regex101.com/r/kIOxuL/1

Reply:
it will work since there will be no "google" in the string but googlebot, at least the list that we have in this PR would work for sure

Reply:
I can't understand works exactly like stripos for me, regarding performance I don't know which method is fastest

Reply:
User-agent can be anything ... just name your user-agent like an ignored bot and and an suffix to name.

E.g. "googlebot" is ignored. You user-agent is called "testgooglebot".  `stripos` would ignore it, `preg_match` does not.

Reply:
I'm sorry, you'll have to forgive me, but I still don't understand 😕


Here's a test I did to be 100% sure: https://reqbin.com/odyy5npi.
```
<?php
$userAgent = 'testgooglebot';
$ignoreAgents[] = 'googlebot';
$ignoreAgents[] = 'bing';

foreach($ignoreAgents as $ignoreAgent) {
	if (stripos($userAgent, $ignoreAgent) !== false) {
		echo 'matched by stripos';
		break;
	}
}
echo PHP_EOL;
$ignoreAgentRegex = implode('|', $ignoreAgents);
if(preg_match("/{$ignoreAgentRegex}/i", $userAgent) !== false){
  echo 'matched by regexp';
}
```

As you can see, if the user agent is called testgooglebot both match it, also stripos not ignore it.
There will never be a real user agent with the string googlebot in it and if preg_match is more efficient, I would keep that